tests/transformers/test_transformers_model_export.py

```diff
@@ -3533,11 +3536,28 @@ def test_qa_pipeline_pyfunc_predict_with_kwargs(small_qa_pipeline):
             "params": parameters,
         }
     )
+    output = mlflow.transformers.generate_signature_output(small_qa_pipeline, data)
     signature_with_params = infer_signature(
         data,
-        mlflow.transformers.generate_signature_output(small_qa_pipeline, data),
+        output,
         parameters,
     )
+    expected_signature = ModelSignature(
+        Schema(
+            [
+                ColSpec(Array(DataType.string), name="question"),
+                ColSpec(Array(DataType.string), name="context"),
+            ]
+        ),
+        Schema([ColSpec(DataType.string)]),
+        ParamSchema(
+            [
+                ParamSpec("top_k", DataType.long, 2),
+                ParamSpec("max_answer_len", DataType.long, 5),
+            ]
+        ),
+    )
+    assert signature_with_params == expected_signature
```

Comment:
Can we add examples of Array type signatures where 'bad' data is passed in that causes the exception handling to trigger to ensure that we have positive capture of what we're trying to prevent users from doing?